src/core/RequestHandler.cpp

```diff
@@ -32,9 +32,25 @@ namespace core {
 		return *this;
 	}
 
-	void RequestHandler::handle(int fd, uint32_t event) {
-		(void)fd;
-		(void)event;
+	// note: the implementation is just temporary
+	void RequestHandler::handle(HandlerContext &ctx) {
+		int fd = ctx.conn.getSocket().getFd();
+		char buffer[1024];
+
+		ssize_t bytesReceived = recv(fd, buffer, sizeof(buffer) - 1, 0);
+		if (bytesReceived < 0) {
+			ctx.conn.close();
+			std::cerr << "Error receiving data" << std::endl;
+			return;
+		}
+		if (bytesReceived == 0) {
+			ctx.conn.close(); //??? why can I close it in read and response? sus
+			return;
+		}
+		buffer[bytesReceived] = '\0';
```

Comment:
*Probably* not strictly necessary. Although I see you read BUFFER_SIZE - 1 so thats prolly okay / safer than without.

Reply:
probably, reading and writing logic is not really implemented yet I was just playing around

Reply:
Huh... don't we need to close in either of these cases? I assumed the FD would still be open locally, even though the client closed its side of it.